### ISSUE TITLE
Workflow with Endpoints in Descriptors provided by Registry

### ISSUE THREAD

**Issue by hofermo** [admin-shell-io/aas-specs-api#295]
# What is missing?

In the tutorial for the AAS API the interaction between the multiple Interfaces is specified:

[Tutorial AAS Part 2 API](https://industrialdigitaltwin.org/en/wp-content/uploads/sites/2/2023/07/2023-07-27_IDTA_Tutorial_V3.0-Specification-AAS-Part-2_API.pdf) 
PDF Page 8
![image](https://github.com/user-attachments/assets/d2db55c4-1f29-4b40-a662-e0bd20673415)

With the current spec it would be possible to have a Submodel Descriptor with an Endpoint that describes an AAS-Registry which does not make sense.

[AAS Specification Part 2 API](https://industrialdigitaltwin.org/wp-content/uploads/2023/06/IDTA-01002-3-0_SpecificationAssetAdministrationShell_Part2_API_.pdf)
PDF Page 82
![image](https://github.com/user-attachments/assets/3fc906d5-f233-460a-9db8-11466ce6d375)
...
 

# How should it be fixed?

These questions arose when we tried to implement Registries in [Mnestix Browser](https://github.com/mnestix/mnestix-browser):

Is there somewhere specified what endpoints a AAS Descriptor and a Submodel Descriptor are allowed to have?
How should it be handled when e.g. an AAS Descriptor has an endpoint with a AAS Registry?
How should it be handled when e.g. an AAS Descriptor has an endpoint to a AAS as well as an endpoint to a AAS Repository?
Where is the source of truth when we have a SM Descriptor with an endpoint "Submodel" and an enpoint "Submodel-Repository"? -> There could be discrepancies when getting data from the Submodel Repo compared to the Submodel endpoint data. What should be higher prioritized?

A diagram or further explanation in the specification would be nice to clear up on how exactly the specified behaviour should look like.

**Comment by BirgitBoss** [comment]
Good point, we did not introduce constraints which values are allowed for Endpoint/interface in which context. 
I would not be backward compatible to do so.

What we started to do is do give advice in the FAQ, e.g. here:
[What are the right attribute values for 'Descriptor/endpoints?](https://github.com/admin-shell-io/questions-and-answers?tab=readme-ov-file#id47)
So we could add some more best practice examples.

These are the described interactions being part of the specificaition themselves:
https://admin-shell-io.github.io/aas-specs-antora/IDTA-01002/v3.1/http-rest-api/interactions.html
From the description it might get clear what is expected although not explicitly specified.
